### ISSUE TITLE
basic.stateswitch has problem with boolean item (iobroker-driver)

### ISSUE THREAD

**Issue by uwe5** [Martin-Gleiss/smartvisu#650]
If the item 'alias.0.bell.state' is of type Boolean, basic.stateswitch will not work, but if the item a number with 1 or 0, it will work. I tried this parameter passing:

`
{{ basic.stateswitch('', 'alias.0.bell.state', '', ['1', '0'],         ['message_bell_door.svg', 'message_bell_door_disabled.svg'], '', ['icon1', 'icon0']) }}
{{ basic.stateswitch('', 'alias.0.bell.state', '', [1, 0],             ['message_bell_door.svg', 'message_bell_door_disabled.svg'], '', ['icon1', 'icon0']) }}
{{ basic.stateswitch('', 'alias.0.bell.state', '', [true, false],      ['message_bell_door.svg', 'message_bell_door_disabled.svg'], '', ['icon1', 'icon0']) }}
`
Tested with the iobroker driver.

Can the basic.stateswitch also work with boolean?

**Comment by wvhn** [comment]
Tested with the smarthomeNG driver, the widget works well with boolean items in all three variants from above.

While smartVISU uses 0 and 1 internally, the driver converts boolean values true and false received from the backend. To convert the values reveiced from iobroker, you could test extending the function "stateChanged in the driver io_iobroker.js
``` 
	stateChanged: function(item, state) {
		var val = state.val;
		if (val === false) 
			val = 0;
		if (val === true) 
			val = 1;
		widget.update(item, val);
	},
```
When you change the item by the stateswitch smartVISU sends "" and "1".  You can test the reaction manually by entering `io.write('alias.0.bell.state', "")` in the browsers developers console and test also "0" and "1".

**Comment by uwe5** [comment]
Thanks very much,
With the change in stateChanged, it works with iobroker.

Merry Christmas

**Comment by wvhn** [comment]
fixed w/ #658
